src/network-uptime/network-dashboard.py

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+import pandas as pd
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.ticker import MaxNLocator
+import pymongo
+import requests
+import streamlit as st
+
+matplotlib.use('Agg')
+
+st.set_page_config(page_title='NETWORK UPTIME AND DOWNTIME', layout="wide")
+st.title('Data report on network uptime and outages over the first quarter, 2022')
+
+
+api_url = 'https://platform.airqo.net/api/v1/devices?tenant=airqo'
+results = requests.get(api_url)
+devices_data = results.json()["devices"]
+
+device_name = devices_name = [data['name'] for data in devices_data]
+selected_device = st.sidebar.selectbox('Device name', device_name)
+
+
+@st.cache
+def load_device_data(device):
+    api = 'https://platform.airqo.net/api/v1/monitor/devices/uptime?tenant=airqo&startDate=2022-01-01T00:00:00.001Z&endDate=2022-03-25T19:00:00.001Z&device_name=' + device
```

Comment:
@uman95 @123MwanjeMike Is there a way the base url can be put in a environment variable??

Reply:
Thanks @NoahNsimbe. I've added the base url as an environment variable.
However @uman95, I have also noticed that  the start and end date are hard coded values. I think there should be another approach where we can get these as inputs from the user and pass them as variables just as with the device.
@NoahNsimbe, do you have any recommendations regarding this?

Reply:
@123MwanjeMike sure, this is work in progress. For now this is  a reprt for the first quarter uptime.

Reply:
Ohh, thanks pointing that out @uman95. I think we can also add the work in progress label